providers/provider_default.go

```diff
@@ -128,8 +131,14 @@ func (p *ProviderData) ValidateSession(ctx context.Context, s *sessions.SessionS
 
 // RefreshSessionIfNeeded should refresh the user's session if required and
 // do nothing if a refresh is not required
-func (p *ProviderData) RefreshSessionIfNeeded(_ context.Context, _ *sessions.SessionState) (bool, error) {
-	return false, nil
+func (p *ProviderData) RefreshSession(_ context.Context, s *sessions.SessionState) (bool, error) {
+	if s == nil {
+		return false, nil
+	}
+
+	// Pretend `RefreshSession` occurred so `ValidateSession` isn't called
+	// on every request after any potential set refresh period elapses.
+	return true, nil
```

Comment:
This is for our few providers that implemented a `ValidateSession` but not `RefreshSession` -- I think in the current implementation if any users set a `--cookie-refresh`, once that time had elapsed, every subsequent request would hit the validate endpoint because they could never successfully refresh and reset the `CreatedAt` time.

Reply:
Ok I think this answers my previous question about what happens for those that don't implement the refresh. I wonder if instead we should return an explicity `ErrNotImplemented` and handle that? It would be more obvious to future contributors IMO

Reply:
I'm confused about why we aren't using a special error here? Why will all providers call this? Surely the error will be passed up to the session middleware which can then handle it?

Reply:
I'll take another peek and see if we can implement with the error, I gave it a whirl and it somehow felt tedious, but I can't remember why 🤔 

The other spots of the interface where we have `ErrNotImplemented` are truly optional codepaths (e.g. `CreateSessionFromToken`) where if someone tried to send a bearer token to a provider that didn't support it, it would error out.

Vs `ValidateSession` which is in the login (and soon refresh) flow for all providers.

Reply:
Latest commit has the shift 👍